### PR TITLE
Np 47357/np 47601 add curator filter

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/common/builder/PartOfQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.jupiter.api.Test;
+
+class PartOfQueryTest {
+
+    @Test
+    void checkPartOfQuery() {
+        var partOfQuery = new PartOfQuery<ResourceParameter>();
+        assertNotNull(
+                partOfQuery.buildMatchAnyValueQuery(
+                        ResourceParameter.CONTRIBUTORS_OF_CHILD, "ewsrdf"));
```
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/common/builder/PartOfQueryTest.java:16]: What is ewsrdf?

**Review by brinxmat — DISMISSED** [review]

File: search-commons/src/test/java/no/unit/nva/search/common/builder/PartOfQueryTest.java (hunk 1 above)
Comment (StigNorland) [reply search-commons/src/test/java/no/unit/nva/search/common/builder/PartOfQueryTest.java:16]: This test is only to satisfy 100% codecoverage, buildMatchAnyValueQuery is never used by PartOfQuery, @JacocoGenerated, doesn't help here...

File: search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java [hunk 2]
```diff
@@ -105,7 +105,12 @@ public ResourceFilter organizationCurationInstitutions(RequestInfo requestInfo)
         if (isSearchingForAllPublications(requestInfo)) {
             return this;
         } else {
-            return organizationCurationInstitutions(requestInfo.getCurrentCustomer());
+            var organization = requestInfo.getCurrentCustomer();
```
Comment (monaullah) [inline search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java:108]: Rename `organization` to `customer`?

File: search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java [hunk 3]
```diff
@@ -105,7 +105,12 @@ public ResourceFilter organizationCurationInstitutions(RequestInfo requestInfo)
...
+            var curationInstitutions =
```
Comment (monaullah) [inline search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java:109]: This is not a list, rename to `curatingInstitution`?

File: search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java [hunk 4]
```diff
@@ -133,15 +138,17 @@ public ResourceFilter organization(URI organization) throws UnauthorizedExceptio
      * @param organization the organization
      * @return ResourceQuery (builder pattern)
      */
-    public ResourceFilter organizationCurationInstitutions(URI organization)
-            throws UnauthorizedException {
+    public ResourceFilter organizationCurationInstitutions(
+            URI organization, URI curationInstitutions) {
```
Comment (monaullah) [inline search-commons/src/main/java/no/unit/nva/search/resource/ResourceFilter.java:142]: Same comment on input param names here

**Review by monaullah — DISMISSED** [review]

File: buildSrc/build.gradle [hunk 5]
```diff
@@ -16,9 +16,9 @@ dependencies {
 }
 
 group = 'com.github.bibsysdev'
-version '1.0-JDK17'
-java.sourceCompatibility = JavaVersion.VERSION_17
-java.targetCompatibility = JavaVersion.VERSION_17
+version '1.0-JDK_21_8.6'
+java.sourceCompatibility = JavaVersion.VERSION_21
```
Comment (brinxmat) [inline buildSrc/build.gradle:20]: This is the old syntax, use 

```
java {
    sourceCompatibility = JavaVersion.VERSION_21
    targetCompatibility = JavaVersion.VERSION_21
}
```

Comment (brinxmat) [reply buildSrc/build.gradle:20]: In another PR, bump pmd version to '7.5.0' in java-conventions

File: template.yaml [hunk 6]
```diff
@@ -8,7 +8,7 @@ Globals:
   Function:
     Timeout: 60
     MemorySize: 1800
-    Runtime: java17
+    Runtime: java21
```
Comment (brinxmat) [inline template.yaml:11]: Is it easier to set this in Globals?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: template.yaml (hunk 6 above)
Comment (StigNorland) [reply template.yaml:11]: se publication...

**Pull request merged**